KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using AngleSharp.Dom.Events;
+using FakeItEasy;
+using KenticoCloud.Delivery.InlineContentItems;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace KenticoCloud.Delivery.Tests
+{
+    public class CodeFirstModelProviderTests
+    {
+        [Fact]
+        // During processing of inline content items, item which detects circular dependency ( A refs B, B refs A ) should resolve resolved item
+        // as if there were no inline content items, which will prevent circular dependency
+        public void RetrievingContentModelWithCircularDependencyDoesNotCycle()
+        {
+            var fakeDeliverClient = A.Fake<IDeliveryClient>();
+            var codeFirstTypeProvider = A.Fake<ICodeFirstTypeProvider>();
+            A.CallTo(() => codeFirstTypeProvider.GetType(A<string>._)).Returns(typeof(ContentItemWithSingleRTE));
+            
+            var processor = new InlineContentItemsProcessor(null, null);
+            processor.RegisterTypeResolver(new RichTextInlineResolver());
+            var retriever = new CodeFirstModelProvider(fakeDeliverClient);
+            retriever.TypeProvider = codeFirstTypeProvider;
+            A.CallTo(() => fakeDeliverClient.InlineContentItemsProcessor).Returns(processor);
+
+            var item = JToken.FromObject(rt1);
+            var modularContent = JToken.FromObject(modularContentObject);
+
+            var result = retriever.GetContentItemModel<ContentItemWithSingleRTE>(item, modularContent);
+
+            Assert.Equal("<span>FirstRT</span><span>SecondRT</span><span>FirstRT</span>", result.RT);
+            Assert.IsType<ContentItemWithSingleRTE>(result);
+        }
+
+        private class ContentItemWithSingleRTE
+        {
+            public string RT { get; set; }
+        }
+
+        private class RichTextInlineResolver: IInlineContentItemsResolver<ContentItemWithSingleRTE>
+        {
+            public string Resolve(ResolvedContentItemData<ContentItemWithSingleRTE> data)
+            {
+                return data.Item.RT;
+            }
+        }
+
+        private static object rt1 = new
```

Comment:
Could you add an extra test for an items that directly reference themselves? RT3->RT3